wicketd/src/update_tracker.rs

```diff
@@ -949,9 +971,12 @@ impl UpdateDriver {
             .map(|response| response.into_inner())
     }
 
-    async fn update_sp(&self, artifact: &ArtifactIdData) -> anyhow::Result<()> {
-        const SP_COMPONENT: &str = SpComponent::SP_ITSELF.const_as_str();
-
+    async fn update_sp_component(
+        &self,
+        artifact: &ArtifactIdData,
+        component: SpComponent,
+    ) -> anyhow::Result<()> {
+        let component_name = component.const_as_str();
```

Comment:
This feels a tiny bit dicey, because `const_as_str` can panic if `component` is malformed. Its docs say 

```
    /// This function should only be used in const contexts when the caller
    /// knows the component is valid (e.g., one of this type's associated
    /// constants); for component names parsed or constructed at runtime, prefer
    /// [`SpComponent::as_str()`] which performs runtime validation.
```

In this case, we're being called with only constants defined in `SpComponent` so everything is fine, but it's not obvious locally that that's the case - we have to check all the callers of this function to know. I'd suggest one of two options:

1. Call `.as_str()` instead and just return an error if a component is invalid
2. Change the function to take `component: &str` and call `.const_as_str()` at each call site where it's obviously correct

Reply:
Yeah that makes sense. I'll probably just switch to 2.

Reply:
Done.